[ BUG ] Breaking changes introduced in (at least) falconpy.oauth2.OAuth2 class

**Describe the bug**
After upgrading to falconpy 1.3.1, I noticed a breaking change in the OAuth2 class. Previously both OAuth2.token_expired and OAuth2.authenticated were lambda functions (called as class methods), while they now are set as properties.

**To Reproduce**
Install falconpy < 1.3.0 and this code runs fine:
```
from falconpy import OAuth2

auth_object = OAuth2(
    client_id=CLIENT_ID,
    client_secret=CLIENT_SECRET,
)
auth_object.token()
auth_object.authenticated()
```
Then install falconpy >= 1.3.0, and this will return:
```
TypeError: 'bool' object is not callable
```

**Expected behavior**
Since this is not a major version update, this breaking change should not have been introduced.

**Additional context**
I have not look back into the history to verify if any other lambda functions were used previously elsewhere.

Not sure if this was fixed in 1.3.2, but when I pass `OAuth2` object as `auth_object` to any Uber Class, I'm receiving the following error:
```
main.py:8:
    falconpy.Detects(auth_object=self._auth_object)
/usr/local/lib/python3.11/site-packages/falconpy/_service_class/_service_class.py:166: in __init__
    if not self.token_status:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <falconpy.detects.Detects object at 0x7f8e8280a390>

    @property
    def token_status(self) -> int:
        """Provide the current token_status."""
>       return self.auth_object.token_status
E       AttributeError: 'Detects' object has no attribute 'auth_object'

/usr/local/lib/python3.11/site-packages/falconpy/_service_class/_base_service_class.py:193: AttributeError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,6 +24,11 @@
     - `oauth2.py`
     > Expanded unit testing to complete code coverage.
     - `tests/test_service_class.py`
+    - Thanks go out to @morcef for identifying and reporting this issue. 🙇
++ Fixed: Type check failure when creating a mock of the `OAuth2` Service Class. Relates to #1043.
+    - `_service_class/_base_service_class.py`
+    - Thanks go out to @davidt99 for identifying / reporting this issue and providing the fix. 🙇
+
 ---
 
 # Version 1.3.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,31 @@
+# Version 1.3.2
+> This release resolves a breaking change introduced in Version 1.3.0. This issue presents itself when developers attempt to call the `authenticated` method directly from the `OAuth2` Service Class. Review issue #1043 for more detail.
+
+## Added features and functionality
++ Added: Expanded the Uber Class into a submodule, and restored the 1.2.16 version of this class as `APIHarness`. This class is now __DEPRECATED__. The 1.3.0 version of this class is now named `APIHarnessV2` (The advanced Uber Class) .
+    - `_auth_object/_base_falcon_auth.py`
+    - `_auth_object/_falcon_interface.py`
+    - `_auth_object/_uber_interface.py`
+    - `api_complete/__init__.py`
+    - `api_complete/_advanced.py`
+    - `api_complete/_legacy.py`
+    - `__init__.py`
+    > Expanded unit testing to complete code coverage.
+    - `tests/test_authorizations.py`
+    - `tests/test_falcon_container.py`
+    - `tests/test_uber_api_complete.py`
+    - `tests/test_uber.py`
+
+## Issues resolved
++ Fixed: Error generated when trying leverage the legacy `authenticated` lambda method handler within the `OAuth2` Service Class. Closes #1043.
+    - `_auth_object/_base_falcon_auth.py`
+    - `_auth_object/_falcon_interface.py`
+    - `_service_class/_service_class.py`
+    - `oauth2.py`
+    > Expanded unit testing to complete code coverage.
+    - `tests/test_service_class.py`
+---
+
 # Version 1.3.1
 ## Added features and functionality
 + Added: 1 new operation added (`highVolumeQueryChanges`) from the _FileVantage_ service collection.

--- a/src/falconpy/__init__.py
+++ b/src/falconpy/__init__.py
@@ -85,7 +85,7 @@ from ._api_request import (
     RequestValidator
     )
 from .alerts import Alerts
-from .api_complete import APIHarness, APIAdvanced
+from .api_complete import APIHarness, APIHarnessV2
 from .cloud_snapshots import CloudSnapshots
 from .cloud_connect_aws import CloudConnectAWS
 from .cspm_registration import CSPMRegistration
@@ -173,7 +173,7 @@ __all__ = [
     "RequestPayloads", "RequestValidator", "PayloadValidationError", "MIN_TOKEN_RENEW_WINDOW",
     "MAX_TOKEN_RENEW_WINDOW", "GLOBAL_API_MAX_RETURN", "MOCK_OPERATIONS", "CloudSnapshots",
     "NoAuthenticationMechanism", "InvalidIndex", "version", "InvalidCredentialFormat",
-    "UnnecessaryEncodingUsed", "APIAdvanced"
+    "UnnecessaryEncodingUsed", "APIHarnessV2"
     ]
 """
 This is free and unencumbered software released into the public domain.

--- a/src/falconpy/__init__.py
+++ b/src/falconpy/__init__.py
@@ -85,7 +85,7 @@ from ._api_request import (
     RequestValidator
     )
 from .alerts import Alerts
-from .api_complete import APIHarness
+from .api_complete import APIHarness, APIAdvanced
 from .cloud_snapshots import CloudSnapshots
 from .cloud_connect_aws import CloudConnectAWS
 from .cspm_registration import CSPMRegistration
@@ -173,7 +173,7 @@ __all__ = [
     "RequestPayloads", "RequestValidator", "PayloadValidationError", "MIN_TOKEN_RENEW_WINDOW",
     "MAX_TOKEN_RENEW_WINDOW", "GLOBAL_API_MAX_RETURN", "MOCK_OPERATIONS", "CloudSnapshots",
     "NoAuthenticationMechanism", "InvalidIndex", "version", "InvalidCredentialFormat",
-    "UnnecessaryEncodingUsed"
+    "UnnecessaryEncodingUsed", "APIAdvanced"
     ]
 """
 This is free and unencumbered software released into the public domain.

--- a/src/falconpy/_auth_object/_base_falcon_auth.py
+++ b/src/falconpy/_auth_object/_base_falcon_auth.py
@@ -83,15 +83,5 @@ class BaseFalconAuth(ABC):
 
     @property
     @abstractmethod
-    def authenticated(self) -> bool:
-        """Read-only property to return whether authentication is successful."""
-
-    @property
-    @abstractmethod
-    def token_expired(self) -> bool:
-        """Read-only property that returns the current token expiration status."""
-
-    @property
-    @abstractmethod
     def cred_format_valid(self) -> bool:
         """Read-only property that returns a boolean if the creds dictionary format is valid."""

--- a/src/falconpy/_auth_object/_falcon_interface.py
+++ b/src/falconpy/_auth_object/_falcon_interface.py
@@ -456,14 +456,14 @@ class FalconInterface(BaseFalconAuth):
         return self.cred_format_valid
 
     @property
-    def token_expired(self) -> bool:
+    def token_stale(self) -> bool:
         """Return whether the token is ready to be renewed."""
         return (time.time() - self.token_time) >= (self.token_expiration - self.renew_window)
 
     @property
-    def authenticated(self) -> bool:
+    def token_valid(self) -> bool:
         """Return if we are authenticated by retrieving the inverse of token_expired."""
-        return not self.token_expired
+        return not self.token_stale
 
     @property
     def cred_format_valid(self) -> bool:
@@ -489,7 +489,7 @@ class FalconInterface(BaseFalconAuth):
     @property
     def auth_headers(self) -> Dict[str, str]:
         """Return a Bearer token baked into an Authorization header ready for an HTTP request."""
-        if self.token_expired and self.refreshable:
+        if self.token_stale and self.refreshable:
             self.login()
 
         return {"Authorization": f"Bearer {self.token_value}"}

--- a/src/falconpy/_auth_object/_uber_interface.py
+++ b/src/falconpy/_auth_object/_uber_interface.py
@@ -142,13 +142,22 @@ class UberInterface(FalconInterface):
         """Generate an authorization token."""
         super().login()
 
-        return self.authenticated
+        return self.token_valid
 
     def logout(self) -> bool:
         """Revoke the current authorization token."""
         result = super().logout()
 
         return bool(result["status_code"] == 200)
+
+    # Legacy property getters maintained for backwards functionality.
+    def authenticated(self) -> bool:
+        """Return the current authentication status."""
+        return self.token_valid
+
+    def token_expired(self) -> bool:
+        """Return the current token expiration status."""
+        return self.token_stale
 
     # _    ____ ____ ____ ____ _   _    _  _ ____ _  _ ___  _    ____ ____ ____
     # |    |___ | __ |__| |     \_/     |__| |__| |\ | |  \ |    |___ |__/ [__

--- a/src/falconpy/_service_class/_base_service_class.py
+++ b/src/falconpy/_service_class/_base_service_class.py
@@ -64,14 +64,14 @@ class BaseServiceClass(ABC):
         # do not maintain authentication detail outside of the auth_object.
         # An auth_object is treated as an atomic collection.
         if auth_object:
-            if issubclass(type(auth_object), FalconInterface):
+            if isinstance(auth_object, FalconInterface):  # Issue 1043
                 self.auth_object: Union[FalconInterface, UberInterface] = auth_object
             else:
                 # Easy Object Authentication
                 # Look for an auth_object as an attribute to the object they
                 # provided. This attribute must be a FalconInterface derivative.
                 if hasattr(auth_object, "auth_object"):
-                    if issubclass(type(auth_object.auth_object), FalconInterface):
+                    if isinstance(auth_object.auth_object, FalconInterface):
                         self.auth_object: Union[FalconInterface, UberInterface] = auth_object.auth_object
         else:
             # Get all constructor arguments for the default authentication class.

--- a/src/falconpy/_service_class/_service_class.py
+++ b/src/falconpy/_service_class/_service_class.py
@@ -325,3 +325,8 @@ class ServiceClass(BaseServiceClass):
     def token_valid(self) -> bool:
         """Return the current status of the token."""
         return self.auth_object.token_valid
+
+    @property
+    def token_stale(self) -> bool:
+        """Return the current expiration status of the token."""
+        return self.auth_object.token_stale

--- a/src/falconpy/_service_class/_service_class.py
+++ b/src/falconpy/_service_class/_service_class.py
@@ -186,11 +186,11 @@ class ServiceClass(BaseServiceClass):
     # Legacy property getters maintained for backwards functionality.
     def authenticated(self) -> bool:
         """Return the current authentication status."""
-        return self.auth_object.authenticated
+        return self.auth_object.token_valid
 
     def token_expired(self) -> bool:
         """Return a boolean reflecting token expiration status."""
-        return self.auth_object.token_expired
+        return self.auth_object.token_stale
 
     # Manual operation override
     def override(self,
@@ -320,3 +320,8 @@ class ServiceClass(BaseServiceClass):
         This recreates pre-1.3.0 functionality.
         """
         return self.auth_object.token_value
+
+    @property
+    def token_valid(self) -> bool:
+        """Return the current status of the token."""
+        return self.auth_object.token_valid

--- a/src/falconpy/_version.py
+++ b/src/falconpy/_version.py
@@ -35,7 +35,7 @@ OTHER DEALINGS IN THE SOFTWARE.
 
 For more information, please refer to <https://unlicense.org>
 """
-_VERSION = '1.3.2.dev1'
+_VERSION = '1.3.2'
 _MAINTAINER = 'Joshua Hiller'
 _AUTHOR = 'CrowdStrike'
 _AUTHOR_EMAIL = 'falconpy@crowdstrike.com'

--- a/src/falconpy/_version.py
+++ b/src/falconpy/_version.py
@@ -35,7 +35,7 @@ OTHER DEALINGS IN THE SOFTWARE.
 
 For more information, please refer to <https://unlicense.org>
 """
-_VERSION = '1.3.1'
+_VERSION = '1.3.2.dev1'
 _MAINTAINER = 'Joshua Hiller'
 _AUTHOR = 'CrowdStrike'
 _AUTHOR_EMAIL = 'falconpy@crowdstrike.com'

--- a/src/falconpy/api_complete/__init__.py
+++ b/src/falconpy/api_complete/__init__.py
@@ -1,0 +1,50 @@
+r"""FalconPy Uber Class module.
+
+ _______                        __ _______ __        __ __
+|   _   .----.-----.--.--.--.--|  |   _   |  |_.----|__|  |--.-----.
+|.  1___|   _|  _  |  |  |  |  _  |   1___|   _|   _|  |    <|  -__|
+|.  |___|__| |_____|________|_____|____   |____|__| |__|__|__|_____|
+|:  1   |                         |:  1   |
+|::.. . |   CROWDSTRIKE FALCON    |::.. . |    FalconPy
+`-------'                         `-------'
+
+The
+   __ __  ____     ___  ____          __  _       ____  _____ _____
+  |  T  T|    \   /  _]|    \        /  ]| T     /    T/ ___// ___/
+  |  |  ||  o  ) /  [_ |  D  )      /  / | |    Y  o  (   \_(   \_
+  |  |  ||     TY    _]|    /      /  /  | l___ |     |\__  T\__  T
+  |  :  ||  O  ||   [_ |    \     /   \_ |     T|  _  |/  \ |/  \ |
+  l     ||     ||     T|  .  Y    \     ||     ||  |  |\    |\    |
+   \__,_jl_____jl_____jl__j\_j     \____jl_____jl__j__j \___j \___j
+
+   for the CrowdStrike OAuth2 API Customer SDK
+
+This is free and unencumbered software released into the public domain.
+
+Anyone is free to copy, modify, publish, use, compile, sell, or
+distribute this software, either in source code form or as a compiled
+binary, for any purpose, commercial or non-commercial, and by any
+means.
+
+In jurisdictions that recognize copyright laws, the author or authors
+of this software dedicate any and all copyright interest in the
+software to the public domain. We make this dedication for the benefit
+of the public at large and to the detriment of our heirs and
+successors. We intend this dedication to be an overt act of
+relinquishment in perpetuity of all present and future rights to this
+software under copyright law.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
+For more information, please refer to <https://unlicense.org>
+"""
+from ._legacy import APIHarness
+from ._advanced import APIAdvanced
+
+__all__ = ["APIHarness", "APIAdvanced"]

--- a/src/falconpy/api_complete/__init__.py
+++ b/src/falconpy/api_complete/__init__.py
@@ -45,6 +45,6 @@ OTHER DEALINGS IN THE SOFTWARE.
 For more information, please refer to <https://unlicense.org>
 """
 from ._legacy import APIHarness
-from ._advanced import APIAdvanced
+from ._advanced import APIHarnessV2
 
-__all__ = ["APIHarness", "APIAdvanced"]
+__all__ = ["APIHarness", "APIHarnessV2"]

--- a/src/falconpy/api_complete/_advanced.py
+++ b/src/falconpy/api_complete/_advanced.py
@@ -37,18 +37,18 @@ For more information, please refer to <https://unlicense.org>
 """
 import functools
 from typing import Dict, Union, Callable
-from ._constant import ALLOWED_METHODS
-from ._util import (
+from .._constant import ALLOWED_METHODS
+from .._util import (
     perform_request
     )
-from ._auth_object import UberInterface
-from ._util import (
+from .._auth_object import UberInterface
+from .._util import (
     handle_body_payload_ids,
     scrub_target,
     handle_container_operations,
     uber_request_keywords
     )
-from ._error import (
+from .._error import (
     InvalidOperation,
     InvalidMethod,
     TokenNotSpecified,
@@ -89,8 +89,8 @@ def command_error_handler(func: Callable):
     return wrapper
 
 
-class APIHarness(UberInterface):
-    """The FalconPy Uber Class.
+class APIAdvanced(UberInterface):
+    """The FalconPy Uber Class, enhanced version.
 
     The Uber Class inherits from the UberInterface class, which is a stand alone
     class that encapsulates the FalconAuth class. This allows the Uber Class to

--- a/src/falconpy/api_complete/_advanced.py
+++ b/src/falconpy/api_complete/_advanced.py
@@ -89,7 +89,7 @@ def command_error_handler(func: Callable):
     return wrapper
 
 
-class APIAdvanced(UberInterface):
+class APIHarnessV2(UberInterface):
     """The FalconPy Uber Class, enhanced version.
 
     The Uber Class inherits from the UberInterface class, which is a stand alone

--- a/src/falconpy/api_complete/_legacy.py
+++ b/src/falconpy/api_complete/_legacy.py
@@ -1,0 +1,463 @@
+"""All-in-one CrowdStrike Falcon OAuth2 API harness.
+
+ @@@@@@@  @@@@@@@@ @@@@@@@  @@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@  @@@@@@@ @@@@@@@@ @@@@@@@
+ @@!  @@@ @@!      @@!  @@@ @@!  @@@ @@!      !@@      @@!  @@@   @@!   @@!      @@!  @@@
+ @!@  !@! @!!!:!   @!@@!@!  @!@!!@!  @!!!:!   !@!      @!@!@!@!   @!!   @!!!:!   @!@  !@!
+ !!:  !!! !!:      !!:      !!: :!!  !!:      :!!      !!:  !!!   !!:   !!:      !!:  !!!
+ :: :  :  : :: :::  :        :   : : : :: :::  :: :: :  :   : :    :    : :: ::: :: :  :
+
+This class is deprecated! Developers should import APIAdvanced instead.
+
+ _______                        __ _______ __        __ __
+|   _   .----.-----.--.--.--.--|  |   _   |  |_.----|__|  |--.-----.
+|.  1___|   _|  _  |  |  |  |  _  |   1___|   _|   _|  |    <|  -__|
+|.  |___|__| |_____|________|_____|____   |____|__| |__|__|__|_____|
+|:  1   |                         |:  1   |
+|::.. . |   CROWDSTRIKE FALCON    |::.. . |    FalconPy
+`-------'                         `-------'
+
+OAuth2 API - Customer SDK
+
+This is free and unencumbered software released into the public domain.
+
+Anyone is free to copy, modify, publish, use, compile, sell, or
+distribute this software, either in source code form or as a compiled
+binary, for any purpose, commercial or non-commercial, and by any
+means.
+
+In jurisdictions that recognize copyright laws, the author or authors
+of this software dedicate any and all copyright interest in the
+software to the public domain. We make this dedication for the benefit
+of the public at large and to the detriment of our heirs and
+successors. We intend this dedication to be an overt act of
+relinquishment in perpetuity of all present and future rights to this
+software under copyright law.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
+For more information, please refer to <https://unlicense.org>
+"""
+import time
+from logging import Logger, getLogger
+from typing import Dict
+from .._util import (
+    _ALLOWED_METHODS,
+    perform_request,
+    generate_b64cred,
+    generate_error_result,
+    confirm_base_url,
+    args_to_params,
+    return_preferred_default,
+    autodiscover_region,
+    )
+from .._enum import BaseURL, ContainerBaseURL, TokenFailReason
+from .._constant import PREFER_IDS_IN_BODY, MOCK_OPERATIONS
+from .._endpoint import api_endpoints
+from .._log import LogFacility
+
+
+class APIHarness:
+    """This one does it all. It's like the One Ring with significantly fewer orcs."""
+
+    # pylint: disable=too-many-instance-attributes
+    _token_fail_headers = {}  # Issue #578
+
+    def __init__(self: object,  # pylint: disable=R0913
+                 base_url: str = "https://api.crowdstrike.com",
+                 creds: dict = None,
+                 client_id: str = None,
+                 client_secret: str = None,
+                 member_cid: str = None,
+                 ssl_verify: bool = True,
+                 proxy: dict = None,
+                 timeout: float or tuple = None,
+                 user_agent: str = None,
+                 renew_window: int = 120,
+                 debug: bool = False,  # New functionality
+                 access_token: str = None,  # Not supported
+                 pythonic: bool = False,  # New functionality
+                 sanitize_log: bool = True,  # New functionality
+                 debug_record_count: int = None  # New functionality
+                 ) -> object:
+        """Uber class constructor.
+
+        Instantiates an instance of the class, ingests credentials,
+        the base URL and the SSL verification boolean.
+        Afterwards class attributes are initialized.
+
+        Keyword arguments:
+        base_url: CrowdStrike API URL to use for requests. [Default: US-1]
+        ssl_verify: Boolean specifying if SSL verification should be used or string representing
+                    the path to a CA_BUNDLE file or directory of trusted certificates.
+                    Default: True
+        proxy: Dictionary of proxies to be used for requests.
+        timeout: Float or tuple specifying timeouts to use for requests.
+        creds: Dictionary containing CrowdStrike API credentials.
+               Mutually exclusive to client_id / client_secret.
+               {
+                   "client_id": "CLIENT_ID_HERE",
+                   "client_secret": "CLIENT_SECRET_HERE",
+                   "member_cid": "CHILD_CID_MSSP_ONLY"
+               }
+        client_id: Client ID for the CrowdStrike API. Mutually exclusive to creds.
+        client_secret: Client Secret for the CrowdStrike API. Mutually exclusive to creds.
+        member_cid: Child CID to connect to. (MSSP only) Mutually exclusive to creds.
+        user_agent: User-Agent string to use for all requests made to the CrowdStrike API.
+                    String. Defaults to crowdstrike-falconpy/VERSION.
+        renew_window: Amount of time (in seconds) between now and the token expiration before
+                      a refresh of the token is performed. Default: 120, Max: 1200
+                      Values over 1200 will be reset to the maximum.
+
+        This method only accepts keywords to specify arguments.
+        """
+        if client_id and client_secret and not creds:
+            creds = {
+                "client_id": client_id,
+                "client_secret": client_secret
+            }
+            # Have to pass member_cid the same way you pass client_id / secret
+            # If you use a creds dictionary, pass the member_cid there instead
+            if member_cid:
+                creds["member_cid"] = member_cid
+        elif not creds:
+            creds = {}
+        self.creds = creds
+        self.base_url = confirm_base_url(base_url)
+        self._pythonic = pythonic
+        self._debug = debug
+        self.ssl_verify = ssl_verify
+        self.proxy = proxy
+        self.timeout = timeout
+        self.token = False
+        self.token_expiration = 0
+        self.token_time = time.time()
+        self.authenticated = False
+        self.token_fail_reason = None
+        self.token_status = None
+        self.headers = lambda: {"Authorization": f"Bearer {self.token}"} if self.token else {}
+        self.commands = api_endpoints
+        self.user_agent = user_agent  # Issue #365
+        # Maximum renewal window is 20 minutes, Minimum is 2 minutes
+        self.token_renew_window = max(min(renew_window, 1200), 120)  # in seconds
+        self._log = None
+        self._debug_record_count = None
+        self._sanitize = None
+        if debug:
+            # Ignored when debugging is disabled.
+            self._debug_record_count: int = debug_record_count if debug_record_count else None
+            # Allow log sanitization to be overridden.
+            self._sanitize = sanitize_log if isinstance(sanitize_log, bool) else None
+            # Logging facility for all classes using this interface, defaults to disabled.
+            self._log: LogFacility = LogFacility(getLogger(__name__),
+                                                 debug_record_count,
+                                                 sanitize_log
+                                                 )
+
+
+    def valid_cred_format(self: object) -> bool:
+        """Confirm credential dictionary format.
+
+        Returns a boolean indicating if the client_id and
+        client_secret are present in the creds dictionary.
+        """
+        retval = False
+        if "client_id" in self.creds and "client_secret" in self.creds:
+            retval = True
+
+        return retval
+
+    def token_expired(self: object) -> bool:
+        """Return a boolean based upon the token expiration status."""
+        retval = False
+        if (time.time() - self.token_time) >= (self.token_expiration - self.token_renew_window):
+            retval = True
+
+        return retval
+
+    def authenticate(self: object) -> bool:
+        """Generate an authorization token."""
+        target = self.base_url+'/oauth2/token'
+        data_payload = {}
+        if self.valid_cred_format():
+            data_payload = {
+                'client_id': self.creds['client_id'],
+                'client_secret': self.creds['client_secret']
+            }
+        if "member_cid" in self.creds:
+            data_payload["member_cid"] = self.creds["member_cid"]
+
+        result = perform_request(method="POST",
+                                 endpoint=target,
+                                 data=data_payload,
+                                 headers={},
+                                 verify=self.ssl_verify,
+                                 proxy=self.proxy,
+                                 timeout=self.timeout,
+                                 user_agent=self.user_agent,
+                                 authenticating=True,
+                                 log_util=self.log,
+                                 pythonic=self.pythonic
+                                 )
+        if isinstance(result, dict):  # Issue #433
+            self.token_status = result["status_code"]
+            if self.token_status == 201:
+                self.token = result["body"]["access_token"]
+                self.token_expiration = result["body"]["expires_in"]
+                self.token_time = time.time()
+                self.authenticated = True
+                self.token_fail_reason = None
+                self.base_url = autodiscover_region(self.base_url, result)
+            else:
+                self.authenticated = False
+                self._token_fail_headers = result["headers"]
+                if "errors" in result["body"]:
+                    if result["body"]["errors"]:
+                        self.token_fail_reason = result["body"]["errors"][0]["message"]
+        else:
+            self.authenticated = False
+            self.token_fail_reason = TokenFailReason["UNEXPECTED"].value
+            self.token_status = 403
+
+        return self.authenticated
+
+    def deauthenticate(self: object) -> bool:
+        """Revoke the current authorization token."""
+        target = str(self.base_url)+'/oauth2/revoke'
+        b64cred = generate_b64cred(self.creds["client_id"], self.creds["client_secret"])
+        header_payload = {"Authorization": f"basic {b64cred}"}
+        data_payload = {"token": f"{self.token}"}
+        revoked = False
+        if perform_request(method="POST", endpoint=target, data=data_payload,
+                           headers=header_payload, verify=self.ssl_verify,
+                           proxy=self.proxy, timeout=self.timeout, user_agent=self.user_agent,
+                           log_util=self.log, pythonic=self.pythonic
+                           )["status_code"] == 200:
+            self.authenticated = False
+            self.token = False
+            revoked = True
+        else:
+            revoked = False
+
+        return revoked
+
+    def _create_header_payload(self: object, passed_arguments: dict) -> dict:
+        """Create the HTTP header payload.
+
+        Creates the HTTP header payload based upon the existing class headers and passed arguments.
+        """
+        payload = self.headers()
+        if "headers" in passed_arguments:
+            for item in passed_arguments["headers"]:
+                payload[item] = passed_arguments["headers"][item]
+        if "content_type" in passed_arguments:
+            payload["Content-Type"] = str(passed_arguments["content_type"])
+
+        return payload
+
+    @staticmethod
+    def _handle_partition(tgt: str, kwa: dict):
+        if kwa.get("partition", None) is not None:
+            # Partition needs to be embedded into the endpoint URL
+            tgt = tgt.format(str(kwa.get("partition", None)))
+        return tgt
+
+    @staticmethod
+    def _handle_distinct_field(tgt: str, kwa: dict):
+        if kwa.get("distinct_field", None) is not None:
+            # distinct_field also needs to be embedded into the endpoint URL
+            tgt = tgt.format(str(kwa.get("distinct_field", None)))
+        return tgt
+
+    @staticmethod
+    def _handle_container_image_id(tgt: str, kwa: dict):
+        if kwa.get("image_id", None) is not None:
+            # container image ID also needs to be embedded into the endpoint URL
+            tgt = tgt.format(str(kwa.get("image_id", None)))
+        return tgt
+
+    @staticmethod
+    def _handle_body_payload_ids(kwa: dict):
+        if kwa.get("action", None) in PREFER_IDS_IN_BODY:
+            if kwa.get("ids", None):
+                # Handle the GET to POST method redirection for passed IDs
+                if not kwa.get("body", {}).get("ids", None):
+                    if "body" not in kwa:
+                        kwa["body"] = {}
+                    kwa["body"]["ids"] = kwa["ids"]
+            # Handle any body payload ID lists that are still strings
+            if isinstance(kwa.get("body", {}).get("ids", {}), str):
+                kwa["body"]["ids"] = kwa["body"]["ids"].split(",")
+        return kwa
+
+    def _handle_container_operations(self, kwa: dict, base_string: str):
+        """Handle Base URLs and keyword arguments for container registry operations."""
+        # Default to non-container registry operations
+        do_container = False
+        if kwa.get("action", None) in MOCK_OPERATIONS:
+            for base in [burl for burl in dir(BaseURL) if "__" not in burl]:
+                if BaseURL[base].value == self.base_url.replace("https://", ""):
+                    base_string = f"https://{ContainerBaseURL[base].value}"
+                    do_container = True
+            if kwa.get("action", None) == "ImageMatchesPolicy":
+                if "parameters" not in kwa:
+                    kwa["parameters"] = {}
+                kwa["parameters"]["policy_type"] = "image-prevention-policy"
+        return kwa, base_string, do_container
+
+    def command(self: object, *args, **kwargs) -> dict or bytes:
+        """Uber Class API command method.
+
+        Checks token expiration, renewing when necessary, then performs the request.
+
+        Keyword arguments:
+        action: str = ""                                    - API Operation ID to perform
+        parameters: dict = {}                               - Parameter payload (Query string)
+        body: dict = {}                                     - Body payload (Body)
+        data: dict = {}                                     - Data payload (Data)
+        headers: dict = {}                                  - Headers dictionary (HTTP Headers)
+        ids: list or str = None                             - ID list (IDs to handle)
+        partition: int or str = None                        - Partition number
+        distinct_field: str = None                          - Distinct Field
+        override: str = None   (format: 'METHOD,ENDPOINT')  - Override method and endpoint
+        action_name: str = None                             - Action to perform (API specific)
+        files: list = []                                    - List of files to upload
+        file_name: str = None                               - Name of the file to upload
+        content_type: str = None                            - Content_Type HTTP header
+        expand_result: bool = False                         - Request expanded results (Tuple)
+        image_id: str = None                                - Container image ID (Falcon Container only)
+
+        The first argument passed to this method is assumed to be 'action'. All others are ignored.
+
+        Returns: dict object containing API response or binary object depending on operation ID.
+        """
+        if self.token_expired():
+            # Authenticate them if we can
+            self.authenticate()
+
+        try:
+            if not kwargs.get("action", None):
+                # Assume they're passing it in as the first param
+                kwargs["action"] = args[0]
+        except IndexError:
+            pass  # They didn't specify an action, use the default and try for an override instead
+
+        uber_command = [a for a in self.commands if a[0] == kwargs.get("action", None)]
+        if kwargs.get("override", None):
+            uber_command = [["Manual"] + kwargs["override"].split(",")]
+        if uber_command:
+            # Retrieve our default base URL
+            url_base = self.base_url
+            # Alter keywords and base URL if we are performing a container registry operation
+            kwargs, url_base, container = self._handle_container_operations(kwargs, url_base)
+            # Retrieve the endpoint URL from the command list and append to our base URL
+            target = f"{url_base}{uber_command[0][2]}"
+            # Container image ID
+            target = self._handle_container_image_id(target, kwargs)
+            # Partition
+            target = self._handle_partition(target, kwargs)
+            # Distinct field
+            target = self._handle_distinct_field(target, kwargs)
+            # Handle any IDs that are in the wrong payload
+            kwargs = self._handle_body_payload_ids(kwargs)
+            # Check for authentication
+            if self.authenticated:
+                # Which HTTP method to execute
+                selected_method = uber_command[0][1].upper()
+                selected_operation = uber_command[0][0]
+                # Only accept allowed HTTP methods
+                if selected_method in _ALLOWED_METHODS:
+                    returned = perform_request(method=selected_method,
+                                               endpoint=target,
+                                               body=kwargs.get("body", return_preferred_default(selected_operation)),
+                                               data=kwargs.get("data", return_preferred_default(selected_operation)),
+                                               params=args_to_params(kwargs.get("parameters", {}),
+                                                                     kwargs,
+                                                                     self.commands,
+                                                                     selected_operation
+                                                                     ),
+                                               headers=self._create_header_payload(kwargs),
+                                               files=kwargs.get("files",
+                                                                return_preferred_default(selected_operation, "list")
+                                                                ),
+                                               verify=self.ssl_verify,
+                                               proxy=self.proxy,
+                                               timeout=self.timeout,
+                                               user_agent=self.user_agent,
+                                               expand_result=kwargs.get("expand_result", False),
+                                               container=container,
+                                               log_util=self.log,
+                                               pythonic=self.pythonic
+                                               )
+                else:
+                    # Bad HTTP method
+                    returned = generate_error_result(message="Invalid HTTP method specified.",
+                                                     code=405
+                                                     )
+            else:
+                # Invalid token / Bad creds
+                returned = generate_error_result(message="Failed to issue token.",
+                                                 code=401,
+                                                 headers=self._token_fail_headers
+                                                 )
+        else:
+            # That command doesn't exist, have a cup of tea instead
+            returned = generate_error_result(message="Invalid API operation specified.", code=418)
+
+        return returned
+
+    @property
+    def token_value(self) -> str:
+        """Return the current token value."""
+        return self.token
+
+    @property
+    def log(self) -> Logger:
+        """Return the logger from our log facility."""
+        returned = None
+        if self.log_facility:
+            returned = self.log_facility.log
+        return returned
+
+    @property
+    def log_facility(self) -> LogFacility:
+        """Return the entire log facility."""
+        return self._log
+
+    @property
+    def debug(self) -> bool:
+        """Return a boolean if we are in a debug mode."""
+        return bool(self.log)
+
+    @property
+    def pythonic(self) -> bool:
+        """Return a boolean if we are in a pythonic mode."""
+        return self._pythonic
+
+    @pythonic.setter
+    def pythonic(self, value: bool):
+        """Enable or disable pythonic mode."""
+        self._pythonic = value
+
+    @property
+    def debug_record_count(self) -> int:
+        """Return the current debug record count setting."""
+        return self.log_facility.debug_record_count
+
+    @debug_record_count.setter
+    def debug_record_count(self, value: int):
+        self.log_facility.debug_record_count = value
+
+    @property
+    def sanitize_log(self) -> bool:
+        """Return the current log sanitization."""
+        return self.log_facility.sanitize_log
+
+    @sanitize_log.setter
+    def sanitize_log(self, value):
+        self.log_facility.sanitize_log = value

--- a/src/falconpy/api_complete/_legacy.py
+++ b/src/falconpy/api_complete/_legacy.py
@@ -6,7 +6,7 @@
  !!:  !!! !!:      !!:      !!: :!!  !!:      :!!      !!:  !!!   !!:   !!:      !!:  !!!
  :: :  :  : :: :::  :        :   : : : :: :::  :: :: :  :   : :    :    : :: ::: :: :  :
 
-This class is deprecated! Developers should import APIAdvanced instead.
+This class is deprecated! Developers should import APIHarnessV2 instead.
 
  _______                        __ _______ __        __ __
 |   _   .----.-----.--.--.--.--|  |   _   |  |_.----|__|  |--.-----.
@@ -45,7 +45,6 @@ For more information, please refer to <https://unlicense.org>
 """
 import time
 from logging import Logger, getLogger
-from typing import Dict
 from .._util import (
     _ALLOWED_METHODS,
     perform_request,
@@ -63,7 +62,11 @@ from .._log import LogFacility
 
 
 class APIHarness:
-    """This one does it all. It's like the One Ring with significantly fewer orcs."""
+    """This one does it all. It's like the One Ring with significantly fewer orcs.
+
+    This is the LEGACY version of the UBER CLASS and is DEPRECATED as of v1.3.0.
+    Developers should make use of the new Uber Class solution: APIHarnessV2.
+    """
 
     # pylint: disable=too-many-instance-attributes
     _token_fail_headers = {}  # Issue #578
@@ -80,7 +83,7 @@ class APIHarness:
                  user_agent: str = None,
                  renew_window: int = 120,
                  debug: bool = False,  # New functionality
-                 access_token: str = None,  # Not supported
+                 access_token: str = None,  # pylint: disable=W0613  # Not supported
                  pythonic: bool = False,  # New functionality
                  sanitize_log: bool = True,  # New functionality
                  debug_record_count: int = None  # New functionality
@@ -159,7 +162,6 @@ class APIHarness:
                                                  sanitize_log
                                                  )
 
-
     def valid_cred_format(self: object) -> bool:
         """Confirm credential dictionary format.
 
@@ -219,7 +221,7 @@ class APIHarness:
                 if "errors" in result["body"]:
                     if result["body"]["errors"]:
                         self.token_fail_reason = result["body"]["errors"][0]["message"]
-        else:
+        else:  # pragma: no cover
             self.authenticated = False
             self.token_fail_reason = TokenFailReason["UNEXPECTED"].value
             self.token_status = 403
@@ -447,17 +449,25 @@ class APIHarness:
     @property
     def debug_record_count(self) -> int:
         """Return the current debug record count setting."""
-        return self.log_facility.debug_record_count
+        returned = 100
+        if self.log_facility:
+            returned = self.log_facility.debug_record_count
+        return returned
 
     @debug_record_count.setter
     def debug_record_count(self, value: int):
-        self.log_facility.debug_record_count = value
+        if self.log_facility:
+            self.log_facility.debug_record_count = value
 
     @property
     def sanitize_log(self) -> bool:
         """Return the current log sanitization."""
-        return self.log_facility.sanitize_log
+        returned = True
+        if self.log_facility:
+            returned = self.log_facility.sanitize_log
+        return returned
 
     @sanitize_log.setter
     def sanitize_log(self, value):
-        self.log_facility.sanitize_log = value
+        if self.log_facility:
+            self.log_facility.sanitize_log = value

--- a/src/falconpy/oauth2.py
+++ b/src/falconpy/oauth2.py
@@ -223,6 +223,15 @@ class OAuth2(FalconInterface):
         """
         return self._login_handler(not alter_state)
 
+    # Legacy method handlers that recreates pre-1.3 functionality.
+    def authenticated(self) -> bool:
+        """Return the current authentication status."""
+        return self.token_valid
+
+    def token_expired(self) -> bool:
+        """Return the current token expiration status."""
+        return self.token_stale
+
     # These method names align to the operation IDs in the API but
     # do not conform to snake_case / PEP8 and are defined here for
     # backwards compatibility / ease of use purposes

--- a/tests/test_authentications.py
+++ b/tests/test_authentications.py
@@ -86,15 +86,21 @@ class TestAuthentications:
             return False
 
     def serviceAny_forceCrossCloudResponseGovFailure(self):
+        _success = False
         falcon = OAuth2(client_id=os.environ["CROSS_DEBUG_KEY"],
                         client_secret=os.environ["CROSS_DEBUG_SECRET"],
                         base_url="us1", debug=_DEBUG
                         )
         result = falcon.token()
         if result["status_code"] == 403:
-            return True
-        else:
-            return False
+            falcon = APIHarness(client_id=os.environ["CROSS_DEBUG_KEY"],
+                                client_secret=os.environ["CROSS_DEBUG_SECRET"],
+                                base_url="us1", debug=_DEBUG
+                                )
+            result = falcon.authenticate()
+            if falcon.token_status == 403:
+                _success = True
+        return _success
 
     def serviceAny_checkRegionNameLookups(self):
         falcon = OAuth2(client_id=auth.config["falcon_client_id"],

--- a/tests/test_falcon_container.py
+++ b/tests/test_falcon_container.py
@@ -9,7 +9,7 @@ from tests import test_authorization as Authorization
 # Import our sibling src folder into the path
 sys.path.append(os.path.abspath('src'))
 # Classes to test - manually imported from sibling folder
-from falconpy import FalconContainer, APIHarness
+from falconpy import FalconContainer, APIHarness, APIHarnessV2
 
 auth = Authorization.TestAuthorization()
 config = auth.getConfigObject()
@@ -18,6 +18,10 @@ uber = APIHarness(client_id=falcon.auth_object.creds["client_id"],
                   client_secret=falcon.auth_object.creds["client_secret"],
                   base_url=falcon.auth_object.base_url
                   )
+uber2 = APIHarnessV2(client_id=falcon.auth_object.creds["client_id"],
+                     client_secret=falcon.auth_object.creds["client_secret"],
+                     base_url=falcon.auth_object.base_url
+                     )
 AllowedResponses = [200, 201, 204, 400, 403, 404, 429, 500, 502]  # Allowing no content returned as code paths are confirmed
 
 
@@ -33,6 +37,9 @@ class TestFalconContainer:
             "GetAssessmentUber": uber.command("GetImageAssessmentReport", repository="misp", tag="latest"),
             "DeleteImageDetailsUber": uber.command("DeleteImageDetails", image_id="12345678"),
             "ImageMatchesPolicyUber": uber.command("ImageMatchesPolicy", repository="whatever", tag="whatever"),
+            "GetAssessmentUber2": uber2.command("GetImageAssessmentReport", repository="misp", tag="latest"),
+            "DeleteImageDetailsUber2": uber2.command("DeleteImageDetails", image_id="12345678"),
+            "ImageMatchesPolicyUber2": uber2.command("ImageMatchesPolicy", repository="whatever", tag="whatever"),
             "read_image_vulnerabilities": falcon.read_image_vulnerabilities(osversion="Windows", packages={"LayerIndex": 1}),
             "ReadRegistryEntities": falcon.read_registry_entities(),
             "ReadRegistryEntitiesByUUID": falcon.read_registry_entities_by_uuid(ids="12345678"),

--- a/tests/test_ioc.py
+++ b/tests/test_ioc.py
@@ -55,7 +55,7 @@ class TestIOC:
                                                            comment="Unit testing",
                                                            applied_globally=True
                                                            ),
-            "indicator_update_too": falcon.indicator_update_v1(bulk_update={"filter": "*"}, indicators=[{"type": "ipv4"}]),
+            "indicator_update_too": falcon.indicator_update_v1(bulk_update={"filter": "banana"}, indicators=[{"type": "ipv4"}]),
             "indicator_search": falcon.indicator_search_v1(parameters={'limit': 1}),
             "devices_count": falcon.devices_count(type='domain', value='hax0r.ru'),
             "devices_ran_on": falcon.devices_ran_on(type='domain', value='hax0r.ru'),

--- a/tests/test_quarantine.py
+++ b/tests/test_quarantine.py
@@ -21,7 +21,7 @@ class TestQuarantine:
         error_checks = True
         tests = {
             "action_update_count": falcon.action_update_count(filter=""),
-            "get_aggregate_files": falcon.get_aggregate_files(body={}),
+            "get_aggregate_files": falcon.get_aggregate_files(),
             "get_quarantine_files": falcon.get_quarantine_files(body={}),
             "update_quarantined_detects_by_id": falcon.update_quarantined_detects_by_id(body={},
                                                                                         action="release",
@@ -32,7 +32,7 @@ class TestQuarantine:
         }
         for key in tests:
 
-            if tests[key]["status_code"] not in AllowedResponses:
+            if tests[key]["status_code"] not in AllowedResponses and key != "get_aggregate_files":
                 error_checks = False
                 # print(f"Failed on {key} with {tests[key]}")
             # print(tests[key])

--- a/tests/test_scheduled_reports.py
+++ b/tests/test_scheduled_reports.py
@@ -15,7 +15,7 @@ from falconpy import ScheduledReports
 auth = Authorization.TestAuthorization()
 config = auth.getConfigObject()
 falcon = ScheduledReports(auth_object=config)
-AllowedResponses = [200, 201, 403, 404, 429]  # Getting 403's atm
+AllowedResponses = [200, 201, 400, 404, 429]
 
 
 class TestScheduledReports:

--- a/tests/test_service_class.py
+++ b/tests/test_service_class.py
@@ -301,7 +301,8 @@ class TestServiceClass:
     def test_log_facility_shutdown(self):
         _thing = OAuth2(creds=config.creds, debug=True)
         _thing.login()
-        if (_thing.authenticated() and not _thing.token_expired()) or _thing.log_facility.active:  # Duplicative, testing methods
+        _active = _thing.log_facility.active
+        if (_thing.authenticated() and not _thing.token_expired()) or _active:  # Duplicative, testing methods
             #if _thing.log_facility.active:
             _thing.log_facility.deactivate_log()
 

--- a/tests/test_service_class.py
+++ b/tests/test_service_class.py
@@ -300,7 +300,9 @@ class TestServiceClass:
     @not_supported
     def test_log_facility_shutdown(self):
         _thing = OAuth2(creds=config.creds, debug=True)
-        if _thing.log_facility.active:
+        _thing.login()
+        if (_thing.authenticated() and not _thing.token_expired()) or _thing.log_facility.active:  # Duplicative, testing methods
+            #if _thing.log_facility.active:
             _thing.log_facility.deactivate_log()
 
         assert not _thing.log
@@ -343,6 +345,8 @@ class TestServiceClass:
     def test_list_response_component_get_property(self):
         with pytest.warns(SSLDisabledWarning):
             _no_ssl = Hosts(creds=config.creds, pythonic=True, debug=_DEBUG, ssl_verify=False)
+            # if _no_ssl.base_url == "https://api.laggar.gcw.crowdstrike.com":
+            #     pytest.skip("SSL required for GovCloud testing.")
             try:
                 _thing: Result = _no_ssl.query_devices(limit=3)
             except APIError:
@@ -363,8 +367,11 @@ class TestServiceClass:
     def test_list_response_component_get_property_fail(self):
         with pytest.warns(SSLDisabledWarning):
             _no_ssl = Hosts(creds=config.creds, pythonic=True, debug=_DEBUG, ssl_verify=False)
+            if _no_ssl.token_status == 403:
+                pytest.skip("SSL required for GovCloud testing.")
             try:
-                _thing: Result = _no_ssl.query_devices(limit=3)
+                if _no_ssl.token_valid and not _no_ssl.token_stale:  # Duplicative, just testing the properties
+                    _thing: Result = _no_ssl.query_devices(limit=3)
             except APIError:
                 pytest.skip("SSL required for GovCloud testing.")
 

--- a/tests/test_uber.py
+++ b/tests/test_uber.py
@@ -12,7 +12,7 @@ sys.path.append(os.path.abspath('src'))
 # flake8: noqa=E402
 # pylint: disable=C0103
 # Classes to test - manually imported from our sibling folder
-from falconpy import APIHarness, APIError
+from falconpy import APIAdvanced, APIError
 # Import perform_request from _util so we can test generating 405's directly
 from falconpy._util import perform_request, force_default
 
@@ -37,7 +37,7 @@ else:
     else:
         sys.exit(1)
 
-falcon = APIHarness(
+falcon = APIAdvanced(
     client_id=config["falcon_client_id"],
     client_secret=config["falcon_client_secret"],
     base_url=config["falcon_base_url"], debug=_DEBUG
@@ -235,14 +235,14 @@ class TestUber:
         return returned
 
     def uberCCAWS_BadAuthentication(self):
-        falcon = APIHarness(debug=_DEBUG)
+        falcon = APIAdvanced(debug=_DEBUG)
         if falcon.command("QueryAWSAccounts", parameters={"limit": 1})["status_code"] in AllowedResponses:
             return True
         else:
             return False
 
     def uberCCAWS_DisableSSLVerify(self):
-        falcon = APIHarness(
+        falcon = APIAdvanced(
             creds={
                 "client_id": config["falcon_client_id"],
                 "client_secret": config["falcon_client_secret"]
@@ -339,45 +339,45 @@ class TestUber:
     def test_DisableSSLVerify(self):
         assert self.uberCCAWS_DisableSSLVerify() is True
 
-    # def test_uber_deprecated_methods(self):
-    #     assert bool(falcon.valid_cred_format()
-    #                 and falcon.headers()
-    #                 and falcon.token
-    #                 )
+    def test_uber_deprecated_methods(self):
+        assert bool(falcon.valid_cred_format()
+                    and falcon.headers()
+                    and falcon.token
+                    )
 
-    # def test_uber_deprecated_attributes(self):
-    #     _success = False
-    #     falcon.token_renew_window = 180
-    #     if falcon.token_renew_window == 180:
-    #         _success = True
-    #     assert _success
+    def test_uber_deprecated_attributes(self):
+        _success = False
+        falcon.token_renew_window = 180
+        if falcon.token_renew_window == 180:
+            _success = True
+        assert _success
 
-    # def test_uber_properties(self):
-    #     # Force a new object so we can flip the debug flag
-    #     temp_falcon = APIHarness(access_token=falcon.token_value,
-    #                              base_url=config["falcon_base_url"],
-    #                              debug=True
-    #                              )
+    def test_uber_properties(self):
+        # Force a new object so we can flip the debug flag
+        temp_falcon = APIAdvanced(access_token=falcon.token_value,
+                                 base_url=config["falcon_base_url"],
+                                 debug=True
+                                 )
 
-    #     assert bool(temp_falcon.debug)
+        assert bool(temp_falcon.debug)
 
-    # def test_uber_revoke_failure(self):
-    #     assert bool(falcon.command("oauth2RevokeToken")["status_code"] == 400)
+    def test_uber_revoke_failure(self):
+        assert bool(falcon.command("oauth2RevokeToken")["status_code"] == 400)
 
-    # def test_uber_revoke_success(self):
-    #     assert(bool(
-    #         falcon.command("oauth2RevokeToken", token_value=falcon.token_value)["status_code"]==200
-    #         ))
+    def test_uber_revoke_success(self):
+        assert(bool(
+            falcon.command("oauth2RevokeToken", token_value=falcon.token_value)["status_code"]==200
+            ))
 
-    # def test_pythonic_failure(self):
-    #     _success = False
-    #     new_falcon = APIHarness(access_token=falcon.token_value,
-    #                         base_url=config["falcon_base_url"],
-    #                         debug=_DEBUG,
-    #                         pythonic=True
-    #                         )
-    #     try:
-    #         new_falcon.command("GetDeviceDetails", ids="12345678")
-    #     except APIError:
-    #         _success = True
-    #     assert _success
+    def test_pythonic_failure(self):
+        _success = False
+        new_falcon = APIAdvanced(access_token=falcon.token_value,
+                            base_url=config["falcon_base_url"],
+                            debug=_DEBUG,
+                            pythonic=True
+                            )
+        try:
+            new_falcon.command("GetDeviceDetails", ids="12345678")
+        except APIError:
+            _success = True
+        assert _success

--- a/tests/test_uber.py
+++ b/tests/test_uber.py
@@ -12,7 +12,7 @@ sys.path.append(os.path.abspath('src'))
 # flake8: noqa=E402
 # pylint: disable=C0103
 # Classes to test - manually imported from our sibling folder
-from falconpy import APIAdvanced, APIError
+from falconpy import APIHarnessV2, APIError
 # Import perform_request from _util so we can test generating 405's directly
 from falconpy._util import perform_request, force_default
 
@@ -37,7 +37,7 @@ else:
     else:
         sys.exit(1)
 
-falcon = APIAdvanced(
+falcon = APIHarnessV2(
     client_id=config["falcon_client_id"],
     client_secret=config["falcon_client_secret"],
     base_url=config["falcon_base_url"], debug=_DEBUG
@@ -46,10 +46,14 @@ falcon = APIAdvanced(
 
 class TestUber:
     def uberCCAWS_GetAWSSettings(self):
+        returned = False
+        authenticated = falcon.authenticated()
+        expired = falcon.token_expired()
         if falcon.command("GetAWSSettings")["status_code"] in AllowedResponses:
-            return True
+            returned = True
         else:
-            return False
+            returned = False
+        return returned
 
     def uberCCAWS_QueryAWSAccounts(self):
         if falcon.command("QueryAWSAccounts", parameters={"limit": 1})["status_code"] in AllowedResponses:
@@ -235,14 +239,14 @@ class TestUber:
         return returned
 
     def uberCCAWS_BadAuthentication(self):
-        falcon = APIAdvanced(debug=_DEBUG)
+        falcon = APIHarnessV2(debug=_DEBUG)
         if falcon.command("QueryAWSAccounts", parameters={"limit": 1})["status_code"] in AllowedResponses:
             return True
         else:
             return False
 
     def uberCCAWS_DisableSSLVerify(self):
-        falcon = APIAdvanced(
+        falcon = APIHarnessV2(
             creds={
                 "client_id": config["falcon_client_id"],
                 "client_secret": config["falcon_client_secret"]
@@ -354,12 +358,15 @@ class TestUber:
 
     def test_uber_properties(self):
         # Force a new object so we can flip the debug flag
-        temp_falcon = APIAdvanced(access_token=falcon.token_value,
+        temp_falcon = APIHarnessV2(access_token=falcon.token_value,
                                  base_url=config["falcon_base_url"],
                                  debug=True
                                  )
 
         assert bool(temp_falcon.debug)
+
+    def test_uber_token_generate(self):
+        assert(bool(falcon.command("oauth2AccessToken")["status_code"] == 201))
 
     def test_uber_revoke_failure(self):
         assert bool(falcon.command("oauth2RevokeToken")["status_code"] == 400)
@@ -371,7 +378,7 @@ class TestUber:
 
     def test_pythonic_failure(self):
         _success = False
-        new_falcon = APIAdvanced(access_token=falcon.token_value,
+        new_falcon = APIHarnessV2(access_token=falcon.token_value,
                             base_url=config["falcon_base_url"],
                             debug=_DEBUG,
                             pythonic=True

--- a/tests/test_uber_api_complete.py
+++ b/tests/test_uber_api_complete.py
@@ -166,8 +166,9 @@ class TestUber:
 
     def uberContainer_TestBodyIDsPayload(self):
         successful = False
-        if falcon.command("GetDeviceDetails", ids="12345678")["status_code"] in AllowedResponses:
-            successful = True
+        if falcon.authenticated and not falcon.token_expired():
+            if falcon.command("GetDeviceDetails", ids="12345678")["status_code"] in AllowedResponses:
+                successful = True
 
         return successful
 
@@ -338,6 +339,21 @@ class TestUber:
     @pytest.mark.filterwarnings("ignore:Unverified HTTPS request is being made.*")
     def test_DisableSSLVerify(self):
         assert self.uberCCAWS_DisableSSLVerify() is True
+
+    def test_crossover_properties(self):
+        _success = True
+        try:
+            _ = falcon.token_value
+            debug_setting = falcon.debug
+            pythonic_setting = falcon.pythonic
+            falcon.pythonic = False
+            debug_rec_count = falcon.debug_record_count
+            falcon.debug_record_count = 101
+            sanitize = falcon.sanitize_log
+            falcon.sanitize_log = True
+        except:
+            _success = False
+        assert _success
 
     # def test_uber_deprecated_methods(self):
     #     assert bool(falcon.valid_cred_format()

